dspace-api/src/main/java/org/dspace/eperson/GroupServiceImpl.java

```diff
@@ -714,27 +703,42 @@ protected void rethinkGroupCache(Context context, boolean flushQueries) throws S
             parent.getValue().addAll(myChildren);
         }
 
-        // empty out group2groupcache table
-        group2GroupCacheDAO.deleteAll(context);
-
-        // write out new one
+        // write out new cache IN MEMORY ONLY and returns it
+        Set<Pair<UUID, UUID>> newCache = new HashSet<>();
         for (Map.Entry<UUID, Set<UUID>> parent : parents.entrySet()) {
             UUID key = parent.getKey();
-
             for (UUID child : parent.getValue()) {
+                newCache.add(Pair.of(key, child));
+            }
+        }
+        return newCache;
+    }
 
-                Group parentGroup = find(context, key);
-                Group childGroup = find(context, child);
 
+    /**
+     * Regenerate the group cache AKA the group2groupcache table in the database -
+     * meant to be called when a group is added or removed from another group
+     *
+     * @param context      The relevant DSpace Context.
+     * @param flushQueries flushQueries Flush all pending queries
+     * @throws SQLException An exception that provides information on a database access error or other errors.
+     */
+    protected void rethinkGroupCache(Context context, boolean flushQueries) throws SQLException {
+        // current cache in the database
+        var oldCache = group2GroupCacheDAO.getCache(context);
```

Comment:
This is one of the first usages of the reserved type name `var` (introduced in Java 10) in the DS Java code base. Are there DS coding guidelines (at least for the Java code base) that describe contexts / cases where using `var` is accepted? Some developers complain about the readability of the Java code when using `var` - especially in diffs where the actual type of a variable (declared as `var`) is not always obvious.

Reply:
@saschaszott : Good point, I'd personally prefer to *not* use `var` as I find it to be vague.  But, we don't have a firm rule on it at this time. 

Nonetheless, I'd rather all the `var` entries in this code be updated to use proper types, as I feel it makes the code more understandable.  But, I'd be interested to hear if there are other opinions.

Reply:
I avoid `var` too.  It seems to me that it would lead to obscure compile errors when refactoring:  instead of a compile error at the assignment (where the expression's type has changed) you get compile errors later when perhaps you cannot see where the type came from.  If I don't know the type of an initializer expression, I use 'int', my IDE complains, and I pick one of its suggestions (or perhaps a generic version thereof).

This discussion should perhaps be happening elsewhere, and the result brought back here.

Reply:
Sorry about that, I often use `var` in our custom code and haven't checked its usage in the DSpace codebase. I have commited the changes.